csclient/csclient.go

```diff
@@ -94,6 +99,15 @@ func New(p Params) *Client {
 			VisitWebPage: p.VisitWebPage,
 		}
 	}
+	if len(p.Auth) > 0 {
+		url, err := url.Parse(p.URL)
+		// A non-nil error here will get caught at request time when we try
+		// to parse the URL, and without a valid URL, the macaroons don't matter
+		// anyway.
+		if err == nil {
```

Comment:
I know this is bad. In discussion with Roger, it was decided that it was better to avoid a breaking change at this time (by forcing New to return an error), so we're just swallowing it for now, though as stated in the comment, the failure to parse would mean that requests will inevitably fail with the same parse failure, at which point, the use of macaroons is moot.